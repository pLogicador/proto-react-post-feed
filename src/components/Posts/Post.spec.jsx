import { render, screen } from "@testing-library/react";
import { Post } from ".";

const props = {
  posts: [
    {
      id: 1,
      title: "title ex1",
      body: "body ex1",
      cover: "img/img1.jpeg",
    },
    {
      id: 2,
      title: "title ex2",
      body: "body ex2",
      cover: "img/img2.jpeg",
    },
    {
      id: 3,
      title: "title ex3",
      body: "body ex3",
      cover: "img/img3.jpeg",
    },
  ],
};

describe("<Posts />", () => {
  it("should render posts", () => {
    render(<Post {...props} />);

    expect(screen.getAllByRole("heading", { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole("img", { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
    expect(screen.getByRole("img", { name: /title ex3/i })).toHaveAttribute(
      "src",
      "img/img3.jpeg"
    );
  });

  /*
  it("should not render posts", () => {
    render(<Post />);

    expect(
      screen.queryByRole("heading", { name: /title/i })
    ).not.toBeInTheDocument();
  });
    */
  it("should match snapshot", () => {
    const { container } = render(<Post {...props} />);
    expect(container).toMatchSnapshot();
  });
});
