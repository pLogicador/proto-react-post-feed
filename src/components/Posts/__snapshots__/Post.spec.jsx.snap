// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div>
  <div
    class="Posts"
  >
    <div
      class="post"
    >
      <img
        alt="title ex1"
        src="img/img1.jpeg"
      />
      <div
        class="post-content"
      >
        <h2>
          1
           
          title ex1
        </h2>
        <p>
          body ex1
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="title ex2"
        src="img/img2.jpeg"
      />
      <div
        class="post-content"
      >
        <h2>
          2
           
          title ex2
        </h2>
        <p>
          body ex2
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="title ex3"
        src="img/img3.jpeg"
      />
      <div
        class="post-content"
      >
        <h2>
          3
           
          title ex3
        </h2>
        <p>
          body ex3
        </p>
      </div>
    </div>
  </div>
</div>
`;
